Orange/widgets/data/owfile.py

```diff
@@ -476,6 +641,24 @@ def workflowEnvChanged(self, key, value, oldvalue):
             self._relocate_recent_files()
             self.set_file_list()
 
+    def apply_domain_edit(self):
+        attributes = []
+        class_vars = []
+        metas = []
+        places = [attributes, class_vars, metas]
+        for (name, tpe, place, vals), orig_var in \
+                zip(self.tablemodel.variables,
+                    chain(self.data.domain.variables, self.data.domain.metas)):
+            if place == 3:
+                continue
+            if name == orig_var.name and tpe == type(orig_var):
+                var = orig_var
+            else:
+                var = tpe(name)
+            places[place].append(var)
+        domain = Domain(attributes, class_vars, metas)
```

Comment:
@kernc @astaric Now what? Read the file with faked headers? This may not work with all readers, and it is undesired if the data is downloaded from the web. The other option is to reinterpret the existing data in numpy. Do we already have something to help with this?